core/src/main/java/pl/netigen/core/netigenapi/NetigenSplashFragment.kt

```diff
@@ -1,4 +1,162 @@
 package pl.netigen.core.netigenapi
 
-class NetigenSplashFragment {
+import android.content.Context
+import android.os.Handler
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.Observer
+import com.google.ads.consent.ConsentInfoUpdateListener
+import com.google.ads.consent.ConsentInformation
+import com.google.ads.consent.ConsentStatus
+import pl.netigen.core.gdpr.ConstGDPR
+import pl.netigen.gdpr.GDPRDialogFragment
+
+abstract class NetigenSplashFragment<ViewModel : NetigenViewModel> : Fragment(), GDPRDialogFragment.GDPRClickListener {
+
+    open lateinit var viewModel: ViewModel
+    var netigenMainActivity: NetigenMainActivity<NetigenViewModel>? = null
+    private var gdprDialogFragment: GDPRDialogFragment? = null
```

Comment:
Why nullable Types?

Reply:
For the same reason getActivity returns a nullable type. And GDPRDialogFragment will be initialized only when ConsentStatus is Unknown, there's no need to initialize it in every other situation